### PR TITLE
Add Wit.ai

### LINKED ISSUE #190: Add a checkbox to switch between keywords and wit.ai
Disable the checkbox if there is any Language detector enabled.

Enabling this checkbox will hide the list of keywords and show the training phrases CRUD that will be used to train the engine.

https://projects.invisionapp.com/d/main#/console/12115078/300980157/preview

Depends on #189

### LINKED ISSUE #191: Add wit.ai setup to the manifest
Expose credentials and traininig information in the manifest

Depends on #190

### LINKED ISSUE #220: Show wit.ai connection status
For #189

### PR DESCRIPTION
**Pull request by aeatencio**
Resolves #190 
Resolves #191 
Resolves #220

Technical debt: #224

### CODE REVIEW

File: app/javascript/ui/KeywordInput.jsx [hunk 1]
```diff
@@ -77,25 +72,98 @@ class KeywordInput extends PureComponent {
...
     }
 
-    return (
-      <div>
-        <Field
+    const useWitAiChange = checked => {
+      if (checked) {
+        openDialog()
+      }
+      if (onUseWitAiChange) {
```
Comment (nekron) [inline app/javascript/ui/KeywordInput.jsx]: This should simply be a required property

File: app/javascript/components/Survey.jsx [hunk 2]
```diff
@@ -63,6 +63,23 @@ class Survey extends Component {
       formIcon = (<FontIcon>file_upload</FontIcon>)
     }
 
+    const addTrainingSentence = () => {
```
Comment (nekron) [inline app/javascript/components/Survey.jsx]: Move this three definitions to the KeywordInput component. Make it so it receives the `updateConfig('training_sentences')` function and uses it to trigger the updates

**Comment by nekron** [comment]
We are missing the connection status badge.

- Let's trigger the credentials popup from there if they are already set. Only trigger it from the checkbox if there are no credentials.

- The badge should only be visible when the checkbox is enabled.

Display 3 states with classes and texts:
- No credentials: grey label "Enable Wit.ai"
- Valid credentials: green label "Connected"
- Invalid credentials: red label "Invalid credentials" (the text from the error message)
  - Reloading the UI when red will result in returning to one of the other two states.

**Comment by aeatencio** [comment]
Thanks for your review, @nekron! I think we are ok now. Please let me know if I missed anything else.

**Pull request merged**